x-pack/test/ingest_manager_api_integration/apis/fleet/agents/reassign.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../../../api_integration/ftr_provider_context';
+import { setupIngest } from './services';
+
+export default function (providerContext: FtrProviderContext) {
+  const { getService } = providerContext;
+  const esArchiver = getService('esArchiver');
+  const supertest = getService('supertest');
+
+  describe('fleet_reassign_agent', () => {
+    setupIngest(providerContext);
+    before(async () => {
+      await esArchiver.loadIfNeeded('fleet/agents');
+    });
+    after(async () => {
+      await esArchiver.unload('fleet/agents');
+    });
+
+    it('should allow to reassign single agent', async () => {
+      await supertest
+        .put(`/api/ingest_manager/fleet/agents/agent1/reassign`)
+        .set('kbn-xsrf', 'xxx')
+        .send({
+          policy_id: 'policy2',
+        })
+        .expect(200);
+      const { body } = await supertest
+        .get(`/api/ingest_manager/fleet/agents/agent1`)
+        .set('kbn-xsrf', 'xxx');
+      expect(body.item.policy_id).to.eql('policy2');
+    });
+
+    it('should throw an error for invalid policy id', async () => {
+      await supertest
+        .put(`/api/ingest_manager/fleet/agents/agent1/reassign`)
+        .set('kbn-xsrf', 'xxx')
+        .send({
+          policy_id: 'INVALID_ID',
+        })
+        .expect(500);
```

Comment:
Seems more like a 4xx error but not a big deal rn.

Reply:
agreed and will defer too, we don't have a super great pattern of handling errors thrown from services in route handlers right now

Reply:
actually apparently we are able to forward the Boom.notFound error :D updated this assertion to 404

Reply:
👍 I did the same in https://github.com/elastic/kibana/pull/77975 & @nchaulet agrees https://github.com/elastic/kibana/pull/77975#discussion_r492054086